client/src/main/java/io/pravega/client/security/auth/ValidJwtTokenHandlingStrategy.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.security.auth;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.gson.Gson;
+import io.pravega.client.stream.impl.Controller;
+import io.pravega.common.Exceptions;
+import io.pravega.common.concurrent.Futures;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.Instant;
+import java.util.Base64;
+import java.util.concurrent.atomic.AtomicReference;
+
+@Slf4j
+public class ValidJwtTokenHandlingStrategy implements DelegationTokenHandlingStrategy {
+
+    /**
+     * Represents the threshold for triggering delegation token refresh.
+     */
+    private static final int REFRESH_THRESHOLD = 10;
+
+    /**
+     * The Controller gRPC client used for interacting with the server.
+     */
+    @Getter(AccessLevel.PROTECTED)
+    private final Controller controllerClient;
+
+    @Getter(AccessLevel.PROTECTED)
+    private final String scopeName;
+
+    @Getter(AccessLevel.PROTECTED)
+    private final String streamName;
+
+    @Getter(AccessLevel.PROTECTED)
+    @Setter(AccessLevel.PROTECTED)
+    private AtomicReference<DelegationToken> delegationToken = new AtomicReference<>();
```

Comment:
I've removed the `@Setter` as it was not needed since the implementation had changed. We still need the `@Getter()` for the `AtomicReference` so that the subclass `NullTokenHandlingStrategy` can make use of it. The `@Getter` has `protected` access level. The object isn't expected to be exposed to callers of the class. Hope that addresses your concern.